File: android/src/main/java/com/godicheol/mymodule/MyModule.java
package com.godicheol.mymodule;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.Promise;

public class MyModule extends ReactContextBaseJavaModule {

    private final ReactApplicationContext reactContext;

    public MyModule(ReactApplicationContext reactContext) {
        super(reactContext);
        this.reactContext = reactContext;
    }

    // To export a module named NativeModules.MyModule.getName
    @Override
    public String getName() {
        return "MyModule";
    }

    // Example method
    // See https://reactnative.dev/docs/native-modules-android
    // To export a module named NativeModules.MyModule.multiply
    @ReactMethod
    public void multiply(double a, double b, Promise promise) {
        promise.resolve(a * b);
    }
}

File: android/src/main/java/com/godicheol/mymodule/MyModule.kt
package com.godicheol.mymodule

import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Callback
import com.facebook.react.bridge.Promise

// edit "MyModule"
class MyModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

  // To export a module named NativeModules.MyModule.getName
  @ReactMethod(isBlockingSynchronousMethod = true)
  override fun getName() = "MyModule" // edit "MyModule"

  // Example method
  // See https://reactnative.dev/docs/native-modules-android
  // To export a module named NativeModules.MyModule.multiply
  @ReactMethod fun multiply(a: Double, b: Double, promise: Promise) {
    promise.resolve(a * b);
  }

}